streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListenerBus.scala

```diff
@@ -17,17 +17,28 @@
 
 package org.apache.spark.streaming.scheduler
 
-import java.util.concurrent.atomic.AtomicBoolean
+import org.apache.spark.scheduler.{LiveListenerBus, SparkListener, SparkListenerEvent}
+import org.apache.spark.util.ListenerBus
 
-import org.apache.spark.Logging
-import org.apache.spark.util.AsynchronousListenerBus
+/**
+ * A Streaming listener bus to forward events in WrappedStreamingListenerEvent to StreamingListeners
+ */
+private[streaming] class StreamingListenerBus(sparkListenerBus: LiveListenerBus)
+  extends SparkListener with ListenerBus[StreamingListener, StreamingListenerEvent] {
 
-/** Asynchronously passes StreamingListenerEvents to registered StreamingListeners. */
-private[spark] class StreamingListenerBus
-  extends AsynchronousListenerBus[StreamingListener, StreamingListenerEvent]("StreamingListenerBus")
-  with Logging {
+  sparkListenerBus.addListener(this)    // for getting callbacks on spark events
 
-  private val logDroppedEvent = new AtomicBoolean(false)
+  def post(event: StreamingListenerEvent) {
```

Comment:
Shouldnt there be a way to remove `this` when the StreamingContext is shutdown. All the added StreamingListener are useless after the StreamingContext has been stopped. Otherwise if someone is starting and stopping StreamingContexts multiple times, all the previous listener will keep getting streaming events of the new one. 

Please add a test for this to make sure that this does not happen.